core/arch/arm/kernel/secure_partition.c

```diff
@@ -341,17 +342,20 @@ static TEE_Result stmm_enter_invoke_cmd(struct tee_ta_session *s,
 
 	mem = &param->u[0].mem;
 	ns_buf_size = mem->size;
-	if (ns_buf_size > spc->ns_comm_buf_size)
-		return TEE_ERROR_SHORT_BUFFER;
+	if (ns_buf_size > spc->ns_comm_buf_size) {
+		mem->size = ns_buf_size;
```

Comment:
`mem->size = spc->ns_comm_buf_size` i think.

Actually there is this `TEEC_ERROR_EXCESS_DATA` return code that looks more appropriate here.
Yet, we should check that `mem->size` output value is well propagated back to caller client when return value is not `TEE_SUCCESS` or `TEE_ERROR_SHORT_BUFFER`.

Reply:
Yes you are right, it should return  spc->ns_comm_buf_size as that's what StMM allows us to use.
I'll have a look and make sure the value is propagated properly.

Reply:
Checked by increasing the u-boot shared memory buffer to 8192. The updated value is indeed propagated correctly